addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -0,0 +1,554 @@
+                const asset = await ajax.loadAsset(this.props.cssReadonlyAssetId);
+
+                if (_avoidDoubleLoad !== avoidDoubleLoad) {
+                    console.warn('Wysiwyg immediate iframe double load detected');
+                    return;
+                }
+                const cwindow = this.iframeRef.el.contentWindow;
+                try {
+                    cwindow.document;
+                } catch (_e) {
+                    return;
+                }
+                cwindow.document
+                    .open("text/html", "replace")
+                    .write(
+                        '<!DOCTYPE html><html>' +
+                        '<head>' +
+                            '<meta charset="utf-8"/>' +
+                            '<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>\n' +
+                            '<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>\n' +
+                            asset.cssLibs.map((cssLib) => {
+                                return '<link type="text/css" rel="stylesheet" href="' + cssLib + '"/>';
+                            }).join('\n') + '\n' +
+                            asset.cssContents.map((cssContent) => {
+                                return '<style type="text/css">' + cssContent + '</style>';
+                            }).join('\n') + '\n' +
+                        '</head>\n' +
+                        '<body class="o_in_iframe o_readonly" style="overflow: hidden;">\n' +
+                            '<div id="iframe_target">' + value + '</div>\n' +
+                            '<script type="text/javascript">' +
+                                'if (window.top.' + this._onUpdateIframeId + ') {' +
+                                    'window.top.' + this._onUpdateIframeId + '(' + _avoidDoubleLoad + ')' +
+                                '}' +
+                            '</script>\n' +
+                        '</body>' +
+                        '</html>');
+
+                const height = cwindow.document.body.scrollHeight;
+                this.iframeRef.el.style.height = Math.max(30, Math.min(height, 500)) + 'px';
+
+                retargetLinks(cwindow.document.body);
+            });
+            // Force the iframe to call the `load` event. Without this line, the
+            // event 'load' might never trigger.
+            this.iframeRef.el.after(this.iframeRef.el);
+
+        });
+        return this.iframePromise;
+    }
+    /**
+     * @private
+     * @param {string} text
+     * @returns {string} the text converted to html
+     */
+    _textToHtml (text) {
+        let value = text || "";
+        try {
+            $(text)[0].innerHTML; // crashes if text isn't html
+        } catch (_e) {
+            if (value.match(/^\s*$/)) {
+                value = '<p><br/></p>';
+            } else {
+                value = "<p>" + value.split(/<br\/?>/).join("<br/></p><p>") + "</p>";
```

Comment:
This function was there before the conversion. I don't know the reason(s) and I think it is out of scope for this PR to investigate.
I don't think constructing `html` is more dangerous in this case because the `value` will be interpreted as html anyway.
This function ensure that the `value` is valid html.
If `value` contain any xss, it is not more dangerous to wrap it in a `<p>`.

Reply:
The function was present before the conversion.
After investigation, this functions seems useless so I've removed it.